### PR TITLE
`r\shared_image`: support `end_of_life_date` `disallowed_disk_types` `[max|min]_recommended_[vcpu_count|memory_in_gb]`

### PR DESCRIPTION
**Pull request by myc2h6o**
Properties are from https://docs.microsoft.com/en-us/azure/virtual-machines/shared-image-galleries?tabs=azure-cli#image-definitions

### CODE REVIEW

**Comment by myc2h6o** [comment]
Test result:
image

File: internal/services/compute/shared_image_resource.go [hunk 1]
```diff
@@ -68,6 +70,25 @@ func resourceSharedImage() *pluginsdk.Resource {
 				}, false),
 			},
 
+			"disallowed_disk_types": {
```
Comment (katbyte) [inline internal/services/compute/shared_image_resource.go:73]: could we change this to 
```suggestion
			"disk_types_not_allowed": {
```

```suggestion
			"disallowed_disk_types": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/compute/shared_image_resource.go (hunk 1 above)
Comment (myc2h6o) [reply internal/services/compute/shared_image_resource.go]: Updated

**Comment by myc2h6o** [comment]
Hi @katbyte thanks for reviewing the pr! I've updated the property name, could you take another look?

**Comment by myc2h6o** [comment]
Updated test result
image

**Review by katbyte — APPROVED** [review]
LGTM 🌻

**Pull request merged**